ui/pages/create-account/connect-hardware/index.js

```diff
@@ -183,7 +184,8 @@ class ConnectHardwareForm extends Component {
           this.setState({ error: U2F_ERROR });
         } else if (
           errorMessage === 'LEDGER_LOCKED' ||
-          errorMessage === 'LEDGER_WRONG_APP'
+          errorMessage === 'LEDGER_WRONG_APP' ||
+          LEDGER_LOCKED_ERROR_CODES.some((code) => errorMessage.includes(code))
```

Comment:
Looks good to me, only comment I could potentially make is that this surrounding `catch` block is pretty gnarly. Could maybe be refactored with `if`'s and `return`'s instead of `else if`'d, and some of the more complex `if` expressions could be variabilised, e.g. this seems more readable to me

```js
const isLedgerLocked = errorMessage === 'LEDGER_LOCKED' 
    || errorMessage === 'LEDGER_WRONG_APP' 
    || LEDGER_LOCKED_ERROR_CODES.some((code) => errorMessage.includes(code));

if(isLedgerLocked) {
   this.setState({
       error: this.context.t('ledgerLocked'),
   });
   return;
}
```

But maybe this is just personal preference 🤔. Up to core MM team if they want this `catch` block refactored.

Reply:
Thanks @andrewpeters9 - happy to refactor or leave as is depending on what the MM team would prefer.